Source/JavaScriptCore/jit/JITPropertyAccess.cpp

```diff
@@ -1437,11 +1437,44 @@ void JIT::emitSlow_op_has_private_brand(const JSInstruction*, Vector<SlowCaseEnt
     emitHasPrivateSlow(AccessType::HasPrivateBrand);
 }
 
-void JIT::emit_op_resolve_scope(const JSInstruction* currentInstruction)
-{
-    auto bytecode = currentInstruction->as<OpResolveScope>();
+#define RESOLVE_SCOPE_CODE_REF(thunk)                                              \
```

Comment:
Can this be a function instead of a macro?

Reply:
The passed thunk function is a template function which cannot be passed as a function pointer. If we need to use function, then we have to create two functions for OpGetFromScope and OpResolveScope each.